piptools/utils.py

```diff
@@ -726,6 +726,8 @@ def _invert_negative_bool_options_in_config(
         # Transform config key to its equivalent in the CLI
         long_option = _convert_to_long_option(key)
         new_key = cli_opts[long_option].name if long_option in cli_opts else key
+        if new_key.startswith("no_"):
+            new_key = new_key[3:]
```

Comment:
I'd probably be more verbose about what that arbitrary constant `3` means:
```suggestion
        negative_option_preffix = "no_"
        if new_key.startswith(negative_option_preffix):
            new_key = new_key[len(negative_option_preffix):]
```